### PR TITLE
fix: close remaining PR #1300 lifecycle regressions

### PR DESCRIPTION
**Pull request by harshitsinghbhandari**
## Summary
- fix legacy merged-session rehydration so merged PR metadata stays merged during canonical synthesis
- validate v2 canonical lifecycle payloads before normalization and fall back to synthesized metadata on malformed payloads
- extend the PR #1300 follow-up note and add regression coverage for merged restore behavior and malformed lifecycle payloads

## Validation
- pnpm build
- pnpm typecheck
- pnpm lint
- pnpm test

## Context
Follow-up to PR #1300 / branch `sessions-redone`. This closes the remaining high-priority core lifecycle issues I found still open during the review follow-up audit.

### CODE REVIEW

File: packages/core/src/lifecycle-state.ts [hunk 1]
```diff
@@ -20,6 +21,84 @@ interface ParseCanonicalLifecycleOptions {
...
+      "process_running",
+      "process_missing",
+      "tmux_missing",
+      "manual_kill_requested",
+      "probe_error",
+    ]),
+    lastObservedAt: TimestampSchema,
+    handle: RuntimeHandleSchema.nullable(),
+    tmuxName: z.string().nullable(),
+  }),
```
[the remark this answers is not in the document]
Comment (harshitsinghbhandari) [reply packages/core/src/lifecycle-state.ts]: Adjusted the v2 payload validator to match the existing tolerant normalizer instead of requiring fully populated nested objects. `session`, `pr`, and `runtime` are now partial/optional, and I added a regression test covering a partial payload that preserves valid session fields while synthesizing the missing sections.

File: packages/core/src/lifecycle-state.ts [hunk 2]
```diff
@@ -20,6 +21,84 @@ interface ParseCanonicalLifecycleOptions {
   createdAt?: Date;
 }
 
+const TimestampSchema = z.string().nullable();
+
+const RuntimeHandleSchema = z.object({
+  id: z.string(),
+  runtimeName: z.string(),
+  data: z.record(z.unknown()),
```
[the remark this answers is not in the document]
Comment (harshitsinghbhandari) [reply packages/core/src/lifecycle-state.ts]: Made `RuntimeHandleSchema` mirror `normalizeRuntimeHandle()` by preprocessing missing or invalid `data` to `{}` instead of rejecting the whole payload. There is now a regression test for a `{ id, runtimeName }` handle with no `data` field so this case stays covered.

**Comment by harshitsinghbhandari** [comment]
Follow-up push for the remaining PR #1300 major fixes is up in db340fad. This pass makes agent-report writes atomic, persists canonical payloads for legacy sessions on read, stabilizes detecting evidence hashes, adds lifecycle-poll observability for swallowed activity/SCM failures, finishes the remaining SessionDetail inline-style cleanup, and updates the review tracker/changeset notes. Validation: targeted core/web/plugin tests passed, full pnpm test passed, pnpm lint passed with existing warnings only, and the repo still has the same unrelated pnpm build / pnpm typecheck failures in notifier/plugin packages outside this diff.

File: packages/web/src/components/SessionDetail.tsx [hunk 3]
```diff
@@ -132,6 +125,39 @@ function activityStateClass(activityLabel: string): string {
...
+      return "session-detail-tone--working";
+    case "var(--color-status-ready)":
+      return "session-detail-tone--ready";
+    case "var(--color-status-attention)":
+      return "session-detail-tone--attention";
+    case "var(--color-status-error)":
+      return "session-detail-tone--error";
+    default:
+      return "session-detail-tone--muted";
+  }
```
[the remark this answers is not in the document]
Comment (harshitsinghbhandari) [reply packages/web/src/components/SessionDetail.tsx:142]: Added the missing idle case so `var(--color-status-idle)` now maps to a dedicated `session-detail-tone--idle` class instead of falling through to muted. That restores the previous idle dot/bar color behavior while keeping the inline-style cleanup intact.

File: packages/core/src/metadata.ts [hunk 4]
```diff
@@ -169,26 +169,50 @@ export function updateMetadata(
...
+  } else if (!options.createIfMissing) {
+    return null;
+  }
+
+  const next = updater({ ...existing });
 
   mkdirSync(dirname(path), { recursive: true });
-  atomicWriteFileSync(path, serializeMetadata(existing));
+  atomicWriteFileSync(path, serializeMetadata(next));
+  return next;
```
[the remark this answers is not in the document]
Comment (harshitsinghbhandari) [reply packages/core/src/metadata.ts]: Normalized mutateMetadata() before both writing and returning, so callers now receive the same record shape that actually survives serialization to disk. I also added a metadata regression test that clears a field via an empty string and asserts the returned object matches readMetadataRaw() exactly.
